IdentityCore/src/oauth2/aad_base/MSIDAADOauth2Factory.m

```diff
@@ -187,6 +188,21 @@ - (BOOL)fillRefreshToken:(MSIDRefreshToken *)token
     return YES;
 }
 
+- (BOOL)fillAppMetadata:(MSIDAppMetadataCacheItem *)metadata
+            fromResponse:(MSIDAADTokenResponse *)response
+           configuration:(MSIDConfiguration *)configuration
+{
+    BOOL result = [super fillAppMetadata:metadata fromResponse:response configuration:configuration];
```

Comment:
please also check that response is MSIDAADTokenResponse

Reply:
response will be MSIDAADTokenResponse as it will be passed in the api

Reply:
I don't think it's guaranteed that the caller will pass MSIDAADTokenResponse, no? We should check explicitly, like we do for credentials I think. There's a helper method to check.

Reply:
added the helper method.